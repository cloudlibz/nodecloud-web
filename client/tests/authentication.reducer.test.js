import * as reducer from "../src/_reducers/authentication.reducer";

describe("reducer", () => {
  describe("LOGIN REQUEST", () => {
    test("sets login to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_LOGIN_REQUEST"
      };
      const nextState = reducer.authentication(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("LOGIN SUCCESS", () => {
    test("sets login to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_LOGIN_SUCCESS"
      };
      const nextState = reducer.authentication(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("LOGIN FAILURE", () => {
    test("sets login to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_LOGIN_FAILURE"
      };
      const nextState = reducer.authentication(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });
});
