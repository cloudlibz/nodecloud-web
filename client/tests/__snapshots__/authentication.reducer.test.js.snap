// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer LOGIN FAILURE sets login to failure 1`] = `Object {}`;

exports[`reducer LOGIN REQUEST sets login to requesting 1`] = `
Object {
  "loggingIn": true,
  "user": undefined,
}
`;

exports[`reducer LOGIN SUCCESS sets login to success 1`] = `
Object {
  "loggedIn": true,
  "user": undefined,
}
`;
